Sources/BrowserServicesKit/PrivacyConfig/ExperimentsDataStore.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  ExperimentsDataStore.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+protocol ExperimentsDataStoring {
+    var experiments: Experiments? { get set }
+}
+
+protocol LocalDataStoring {
+    func data(forKey defaultName: String) -> Data?
+    func set(_ value: Any?, forKey defaultName: String)
+}
+
+struct ExperimentsDataStore: ExperimentsDataStoring {
+    private let localDataStoring: LocalDataStoring
+    private let experimentsDataKey = "ExperimentsData"
+    private let queue = DispatchQueue(label: "com.experimentManager.queue")
```

Comment:
did you consider using actor instead of struct not to use queue for this?

Reply:
It seemed cleaner this way…